vm/array_test.go

```diff
@@ -140,16 +140,214 @@ func TestArrayIndex(t *testing.T) {
+}
+
+func TestArrayIndexWithSuccessiveValues(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected []interface{}
+	}{
+		{`
+			a = [1, 2, 3, 4, 5]
+			a[1, 1]
+		`, []interface{}{2}},
+		{`
+			a = [1, 2, 3, 4, 5]
+			a[1, 3]
+		`, []interface{}{2, 3, 4}},
+		{`
+			a = [1, 2, 3, 4, 5]
+			a[1, 5]
+		`, []interface{}{2, 3, 4, 5}},
+		{`
+			a = [1, 2, 3, 4, 5]
+			a[-5, 1]
+		`, []interface{}{1}},
+		{`
+			a = [1, 2, 3, 4, 5]
+			a[-5, 5]
+		`, []interface{}{1, 2, 3, 4, 5}},
+		{`
+			a = [1, 2, 3, 4, 5]
+			a[1, 10]
+		`, []interface{}{2, 3, 4, 5}},
+		{`
+			a = [1, 2, 3, 4, 5]
+			a[3, 1]
+		`, []interface{}{4}},
+		{`
+			a = [1, 2, 3, 4, 5]
+			a[3, 2]
+		`, []interface{}{4, 5}},
+		{`
+			a = [1, 2, 3, 4, 5]
+			a[3, 3]
+		`, []interface{}{4, 5}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[3, 0]
+		`, []interface{}{}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[3, 1]
+		`, []interface{}{4}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[3, 3]
+		`, []interface{}{4, 5, 6}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[3, 5]
+		`, []interface{}{4, 5, 6, 7, 8}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[3, 7]
+		`, []interface{}{4, 5, 6, 7, 8, 9, 10}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[3, 10]
+		`, []interface{}{4, 5, 6, 7, 8, 9, 10}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[-7, 1]
+		`, []interface{}{4}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[-7, 3]
+		`, []interface{}{4, 5, 6}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[-7, 5]
+		`, []interface{}{4, 5, 6, 7, 8}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[-7, 7]
+		`, []interface{}{4, 5, 6, 7, 8, 9, 10}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[-7, 10]
+		`, []interface{}{4, 5, 6, 7, 8, 9, 10}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[2, 3] = [1, 2, 3, 4, 5]
+		`, []interface{}{1, 2, 3, 4, 5}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[2, 3] = [1, 2, 3, 4, 5]
+			a
+		`, []interface{}{1, 2, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[1, 7] = [1, 2, 3]
+			a
+		`, []interface{}{1, 1, 2, 3, 9, 10}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[-5, 3] = [2, 4, 6, 8, 10]
+			a
+		`, []interface{}{1, 2, 3, 4, 5, 2, 4, 6, 8, 10, 9, 10}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[5, 6] = 123
+			a
+		`, []interface{}{1, 2, 3, 4, 5, 123}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[5, 0] = 123
+			a
+		`, []interface{}{1, 2, 3, 4, 5, 123, 6, 7, 8, 9, 10}},
+		{`
+			a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+			a[-7, 2] = "@Maxwell-Alexius is solving issue #403"
+			a
+		`, []interface{}{1, 2, 3, "@Maxwell-Alexius is solving issue #403", 6, 7, 8, 9, 10}},
+		{`
+			a = [1, 2, 3, 4, 5]
+			a[5, 123] = 123
+			a
+		`, []interface{}{1, 2, 3, 4, 5, 123}},
+		{`
+			a = [1, 2, 3, 4, 5]
+			a[5, 123] = [1, 2, 3]
+			a
+		`, []interface{}{1, 2, 3, 4, 5, 1, 2, 3}},
+		{`
+			a = [1, 2, 3, 4, 5]
```

Comment:
We should document this kind of cases, can you add some comments (or references) here to explain that?

Reply:
Alright, I shall document these behaviour in these days, took me a bit time to clarify these bizarre cases

Reply:
I added to the `vm/array.go` part